cmd/soroban-cli/src/commands/network/start.rs

```diff
@@ -0,0 +1,151 @@
+/// -p <`FROM_PORT`>:<`TO_PORT`> - this maps the port from the container to the host machine. By default, the port is 8000.
+/// -d - this runs the container in detached mode, so that it runs in the background
+
+// By default, without any optional arguments, the following docker command will run:
+// docker run --rm -p 8000:8000 --name stellar stellar/quickstart:testing --testnet --enable-soroban-rpc
+
+#[derive(Debug, clap::Parser, Clone)]
+pub struct Cmd {
+    /// Network to start, e.g. local, testnet, futurenet, pubnet
+    pub network: String,
+
+    /// optional argument to override the default docker image tag for the given network
+    #[arg(short = 't', long)]
+    pub image_tag_override: Option<String>,
+
+    /// optional argument to turn off soroban rpc
+    #[arg(short = 'r', long)]
+    pub disable_soroban_rpc: bool,
+
+    // option argument to specify the protocol version for the local network only
+    #[arg(short = 'p', long)]
+    pub protocol_version: Option<String>,
+
+    // option argument to specify the limits for the local network only
+    #[arg(short = 'l', long)]
+    pub limit: Option<String>,
+
+    // optional arguments to pass to the docker run command
+    #[arg(last = true, id = "DOCKER_RUN_ARGS")]
+    pub slop: Vec<String>,
+}
+
+impl Cmd {
+    pub fn run(&self) -> Result<(), Error> {
+        println!("Starting {} network", &self.network);
+
+        let docker_command = build_docker_command(self);
+
+        run_docker_command(&docker_command);
+        Ok(())
+    }
+}
+
+fn run_docker_command(docker_command: &str) {
+    println!("Running docker command: `{docker_command}`");
+    let mut cmd = Command::new("sh")
+        .args(["-c", &docker_command])
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .spawn()
+        .unwrap();
+
+    let status = cmd.wait();
+    if status.is_err() {
+        println!("Exited with status {status:?}");
+    }
+}
+
+fn build_docker_command(cmd: &Cmd) -> String {
+    let image = get_image_name(cmd);
+    let container_name = get_container_name(cmd);
+    let port_mapping = get_port_mapping(cmd);
+    let protocol_version = get_protocol_version_arg(cmd);
+    let limits = get_limits_arg(cmd);
+
+    let docker_command =
+        format!(
+        "docker run --rm {slop} {port} {container_name} {image} --{network} {enable_soroban_rpc} {protocol_version} {limits}",
+        port = format_args!("-p {port_mapping}"),
+        container_name = format_args!("--name {container_name}"),
+        image = image,
+        network = cmd.network,
+        slop = cmd.slop.join(" "),
+        enable_soroban_rpc = if cmd.disable_soroban_rpc { "" } else { "--enable-soroban-rpc" },
+        protocol_version = protocol_version,
+        limits = limits,
+    );
+
+    docker_command
+}
+
+fn get_image_name(cmd: &Cmd) -> String {
+    // this can be overriden with the `-t` flag
+    let mut image_tag = match cmd.network.as_str() {
```

Comment:
@chadoh and I discussed that it is important to make sure that at any given time the `soroban network start` command will be compatible with a version of the quickstart image, for each network.

one idea we had was to hardcode specific image tag digests, but I wonder if this may be error-prone since it would be easy to forget to update the sha for each release. 

also, as I was digging into how the quickstart images are built and pushed to docker hub, I realized that `latest`, `testing` and `soroban-dev` seem to be pushed to docker hub on every push to main, even if the underlying image is the same, iIwonder if this still affects the digest. 

Anyway, I could use some help with this!

Reply:
Another thing I noticed as I was looking through the quickstart repo is that `latest`, `testing` and `soroban-dev` all seem to be using the same versions of xdr, core, horizon, etc. Does this mean that `[latest](https://github.com/stellar/quickstart/blob/master/.github/workflows/build-latest.yml)` is running protocol 20?

Reply:
Latest has the software that supports protocol 20, but the public network (pubnet/mainnet) is running protocol 19. stellar-core knows how to run with each protocol version and when the validators vote on running protocol 20 the functionality for that protocol will become enabled.

Reply:
Thanks! That's very helpful 👍